ingestion/src/metadata/orm_profiler/profiler/handle_partition.py

```diff
@@ -178,16 +179,15 @@ def handle_and_execute(_self, *args, **kwargs):
                         .filter(partition_filter)
                         .cte(f"{_self.table.__tablename__}_rnd")
                     )
-
-                query_results = (
-                    _self._build_query(*args, **kwargs)
-                    .select_from(_self._sample if self.sampled else _self.table)
-                    .filter(partition_filter)
+                query_results = _self._build_query(*args, **kwargs).select_from(
+                    _self._sample if self.sampled else _self.table
                 )
-                if self.first:
-                    return query_results.first()
-                return query_results.all()
-
+                if partition_field in {"_PARTITIONDATE", "_PARTITIONTIME"}:
+                    if not self.sampled:
```

Comment:
Why do we have this here?

Reply:
so {"_PARTITIONDATE", "_PARTITIONTIME"} and the other fields like column names all differ when the queries are being formed..

if `self.sampled` is `True`, the query it forms has couple of syntax issue where `Where` clause repeats and gives us a syntax error, that's why added a check that if `sampled` is `True`, we don't add a `where` as there already is one present.


Paritioned and sampled throws syntax error..that's why the conditions

Reply:
> if self.sampled is True, the query it forms has couple of syntax issue where Where clause repeats and gives us a syntax error, that's why added a check that if sampled is True, we don't add a where as there already is one present.

^ That probably means there is something to fix somewhere else.